pkgs/by-name/st/stash/package.nix

```
@@ -0,0 +1,52 @@
+{
+  lib,
+  stdenv,
+  fetchurl,
+}:
+let
+
+  version = "0.25.1";
+
+  sources = {
+    x86_64-linux = {
+      url = "https://github.com/stashapp/stash/releases/download/v${version}/stash-linux";
+      hash = "sha256-Rb4x6iKx6T9NPuWWDbNaz+35XPzLqZzSm0psv+k2Gw4=";
+    };
+    aarch64-linux = {
+      url = "https://github.com/stashapp/stash/releases/download/v${version}/stash-linux-arm64v8";
+      hash = "sha256-6qPyIYKFkhmBNO47w9E91FSKlByepBOnl0MNJighGSc=";
+    };
+    x86_64-darwin = {
+      url = "https://github.com/stashapp/stash/releases/download/v${version}/stash-macos";
+      hash = "sha256-W8+rgqWUDTOB8ykGO2GL9tKEjaDXdx9LpFg0TAtJsxM=";
+    };
+  };
+in
+stdenv.mkDerivation (finalAttrs: {
+  inherit version;
+
+  pname = "stash";
+
+  src = fetchurl {
+    inherit (sources.${stdenv.system} or (throw "Unsupported system: ${stdenv.system}")) url hash;
```

You shouldn't need to throw an error here since `meta.platforms` will prevent platforms that are unsupported from trying to build this.
```suggestion
    inherit (sources.${stdenv.system}) url hash;
```

---

Indeed, nice.

---

Thanks, Done